### PR TITLE
Add ros2 utils

### PR DESCRIPTION
**Pull request by Tadinu**
* `UROS2Publisher` add `static CreatePublisher()` for custom type publisher/message
* `ROS2Node.h` add `RR_ROS2_SUBSCRIBE_TO_TOPIC`
* `rclcUtilities.h` Add `UROS2Utils` for ros <-> ue type conversion, used by UE-adapter gen source

### CODE REVIEW

File: Source/rclUE/Public/rclcUtilities.h [hunk 1]
```diff
@@ -139,3 +147,113 @@ static const TMap<TEnumAsByte<UROS2QoS>, rmw_qos_profile_t> QoS_LUT = {
     {UROS2QoS::ParameterEvents, rmw_qos_profile_parameter_events},
     {UROS2QoS::System, rmw_qos_profile_system_default},
     {UROS2QoS::UnknownQoS, rmw_qos_profile_unknown}};
+
+UCLASS()
+class UROS2Utils : public UBlueprintFunctionLibrary
```
Comment (yuokamoto) [inline Source/rclUE/Public/rclcUtilities.h:152]: where this class used?

File: Source/rclUE/Public/rclcUtilities.h [hunk 2]
```diff
@@ -139,3 +147,113 @@ static const TMap<TEnumAsByte<UROS2QoS>, rmw_qos_profile_t> QoS_LUT = {
...
+    {
+        int32 totalCapacity = 0;
+        for (const auto& str : InStrList)
+        {
+            totalCapacity += GetStringRequiredCapacity(str);
+        }
+        return totalCapacity;
+    }
+
+    static void SetStringProp(rosidl_runtime_c__String& OutStr, const FString& InStr)
```
Comment (tetorea) [inline Source/rclUE/Public/rclcUtilities.h]: 'Prop' is for Properties?
Putting the 'In' variable before the 'Out' one in the parameter list seems more logical to me. Or is this a common way to do in ros or ue code?

File: Source/rclUE/Public/ROS2Node.h [hunk 3]
```diff
@@ -35,17 +37,30 @@ DECLARE_DYNAMIC_DELEGATE_OneParam(FActionCallback, UROS2GenericAction*, InAction
 DECLARE_DELEGATE(FSimpleCallback);
 
 /**
- * @brief Helper structs which is components of the node and should register to 
+ * @brief RR_ROS2_SUBSCRIBE_TO_TOPIC
+ * FSubscriptionCallback is of dynamic delegate type to be serializable for BP use
+ * FSubscriptionCallback::BindDynamic is a macro, instead of a function.
+ * Thus InCallback can only be a direct UFUNCTION() method & cannot be used as typed param!
```
Comment (tetorea) [inline Source/rclUE/Public/ROS2Node.h:43]: Compiler warns if InCallback is not a direct UFUNCTION ?

Comment (Tadinu) [reply Source/rclUE/Public/ROS2Node.h:43]: Yeah, it is quite a cruft that for BP-usable dynamic delegate [binding](https://github.com/EpicGames/UnrealEngine/blob/release/Engine/Source/Runtime/Core/Public/Delegates/Delegate.h#L420) is written in a macro, which manually extracts the literal function name then store it [here](https://github.com/EpicGames/UnrealEngine/blob/release/Engine/Source/Runtime/Core/Public/Delegates/DelegateSignatureImpl.inl#L1038)
-> As such, in order to implement a type-based method, we need to hook up directly to `__Internal_BindDynamic`, which is not recommended & have to pass the literal function name ("Foo") along with the method pointer (`&AClass::Foo`)

File: Source/rclUE/Public/rclcUtilities.h (hunk 1 above)
Comment (Tadinu) [reply Source/rclUE/Public/rclcUtilities.h:152]: Yeah, for UE-ROS adapter src as in [RapyutaSimInternal's SOOTBALL_SIM](https://github.com/rapyuta-robotics/RapyutaSimulationInternal/blob/SOOTBALL_SIM/Source/RapyutaSimInternal/Public/Sootball/ROS2/Msgs/ROS2OrderMsg.h#L102)

**Review by yuokamoto — APPROVED** [review]

**Review by tetorea — APPROVED** [review]
Working with my project

**Pull request merged**